tests/symbol_dependent_transients_test.py

```diff
@@ -0,0 +1,163 @@
+    body2_state.add_nedge(rednode, write_tmp2, dace.Memlet('tmp2[0]'))
+
+    read_tmp2 = body3_state.add_read('tmp2')
+    write_b = body3_state.add_write('B')
+    body3_state.add_nedge(read_tmp2, write_b, dace.Memlet('B[i]'))
+
+    return sdfg
+
+
+def test_symbol_dependent_heap_array():
+    A = np.random.randn(10, 10, 10)
+    B = np.ndarray(10, dtype=np.float64)
+    sdfg = _make_sdfg("symbol_dependent_heap_array")
+    # Compile manually to avoid strict transformations
+    sdfg_exec = sdfg.compile()
+    sdfg_exec(A=A, B=B, N=10)
+    B_ref = np.ndarray(10, dtype=np.float64)
+    for i in range(10):
+        tmp = A[2:-2, 2:-2, i:]
+        B_ref[i] = np.sum(tmp)
+    assert (np.allclose(B, B_ref))
+
+
+def test_symbol_dependent_register_array():
+    A = np.random.randn(10, 10, 10)
+    B = np.ndarray(10, dtype=np.float64)
+    sdfg = _make_sdfg("symbol_dependent_register_array",
+                      storage=dace.dtypes.StorageType.Register)
+    # Compile manually to avoid strict transformations
+    sdfg_exec = sdfg.compile()
+    sdfg_exec(A=A, B=B, N=10)
+    B_ref = np.ndarray(10, dtype=np.float64)
+    for i in range(10):
+        tmp = A[2:-2, 2:-2, i:]
+        B_ref[i] = np.sum(tmp)
+    assert (np.allclose(B, B_ref))
+
+
+def test_symbol_dependent_threadlocal_array():
+    A = np.random.randn(10, 10, 10)
+    B = np.ndarray(10, dtype=np.float64)
+    sdfg = _make_sdfg("symbol_dependent_threadlocal_array",
+                      storage=dace.dtypes.StorageType.CPU_ThreadLocal)
+    # Compile manually to avoid strict transformations
+    sdfg_exec = sdfg.compile()
+    sdfg_exec(A=A, B=B, N=10)
+    B_ref = np.ndarray(10, dtype=np.float64)
+    for i in range(10):
+        tmp = A[2:-2, 2:-2, i:]
+        B_ref[i] = np.sum(tmp)
+    assert (np.allclose(B, B_ref))
+
+
+@pytest.mark.gpu
+def test_symbol_dependent_gpu_global_array():
+    A = np.random.randn(10, 10, 10)
+    B = np.ndarray(10, dtype=np.float64)
+    sdfg = _make_sdfg("symbol_dependent_gpu_global_array",
+                      storage=dace.dtypes.StorageType.GPU_Global)
+    # Compile manually to avoid strict transformations
+    sdfg_exec = sdfg.compile()
+    sdfg_exec(A=A, B=B, N=10)
+    B_ref = np.ndarray(10, dtype=np.float64)
+    for i in range(10):
+        tmp = A[2:-2, 2:-2, i:]
+        B_ref[i] = np.sum(tmp)
+    assert (np.allclose(B, B_ref))
+
+
+@pytest.mark.gpu
+def test_symbol_dependent_pinned_array():
+    A = np.random.randn(10, 10, 10)
+    B = np.ndarray(10, dtype=np.float64)
+    sdfg = _make_sdfg("symbol_dependent_pinned_array",
+                      storage=dace.dtypes.StorageType.CPU_Pinned)
+    # Compile manually to avoid strict transformations
+    sdfg_exec = sdfg.compile()
+    sdfg_exec(A=A, B=B, N=10)
+    B_ref = np.ndarray(10, dtype=np.float64)
+    for i in range(10):
+        tmp = A[2:-2, 2:-2, i:]
+        B_ref[i] = np.sum(tmp)
+    assert (np.allclose(B, B_ref))
+
+
+@pytest.mark.skip
```

Comment:
Does this pass?

Reply:
No, but this is probably because of the reasons we discussed. This must always be overapproximated for FPGAs. I am disabling it and I will think of a better test later.